stylix/mk-target.nix

```diff
@@ -0,0 +1,219 @@
+/**
+  Provides a consistent target interface, minimizing boilerplate and
+  automatically safeguarding declarations related to disabled options.
+
+  # Type
+
+  ```
+  mkTarget :: AttrSet -> ModuleBody
+  ```
+
+  # Examples
+
+  This function returns the module body, not an actual module.
+
+  The `modules/«MODULE»/«PLATFORM».nix` modules should use this function as
+  follows:
+
+      { mkTarget, lib... }:
+      mkTarget {
+        name = "«name»";
+        humanName = "«human readable name»";
+
+        generalConfig =
+          lib.mkIf complexCondition {
+            home.packages = [ pkgs.hello ];
+          };
+
+        configElements = [
+          { programs.«name».theme.name = "stylix"; }
+
+          (
+            { colors }:
+            {
+              programs.«name».theme.background = colors.base00;
+            }
+          )
+
+          (
+            { fonts }:
+            {
+              programs.«name».font.name = fonts.monospace.name;
+            }
+          )
+        ];
+      }
+
+  Note that `mkTarget` is provided in the same place as module arguments.
```

Comment:
This sentence should not be in the "Exmaples" section. Moving it to the synopsis is also not good. Maybe we should use a dedicated section, like "Environment"?